### PR TITLE
[spirv] Remove code for deprecated Linalg on buffers path

### PR DESCRIPTION
**Pull request by antiagainst**
Here we go! :) This commit removes code for deprecated Linalg
on buffers path and command-line options that are default now.

### CODE REVIEW

**Comment by antiagainst** [comment]
Sending out PR for now. Need to do more testing internally to make sure everything is fine.

**Comment by benvanik** [comment]
Nice! Thanks for doing this Lei!
Once this lands I'll go clean up some of the HAL stuff related to the split dispatch regions that were hanging around for this code path.

**Review by benvanik — APPROVED** [review]

File: iree/compiler/Conversion/LinalgToSPIRV/ConvertToGPUPass.cpp [hunk 1]
```diff
@@ -601,15 +592,14 @@ LogicalResult mapLinalgOpToLocalInvocationIdImpl<linalg::CopyOp>(
...
-      : OpConversionPattern<LinalgOpTy>(context, benefit),
-        usingLinalgOnTensorsPath(usingLinalgOnTensorsPath) {}
+      : OpConversionPattern<LinalgOpTy>(context, benefit) {}
+
   LogicalResult matchAndRewrite(
       LinalgOpTy linalgOp, ArrayRef<Value> operands,
       ConversionPatternRewriter &rewriter) const override {
-    if (failed(mapLinalgOpToLocalInvocationIdImpl(linalgOp, operands, rewriter,
-                                                  usingLinalgOnTensorsPath)))
+    if (failed(mapLinalgOpToLocalInvocationIdImpl(
```
Comment (MaheshRavishankar) [inline iree/compiler/Conversion/LinalgToSPIRV/ConvertToGPUPass.cpp:601]: Actually, could you just follow the call stack here and remove the arguments as its passed down. I think you can remove it everywhere upto the function at line 441 here.

**Review by MaheshRavishankar — CHANGES_REQUESTED** [review]
Awesome! I think removing `SplitDispatchFunctionPass` was the reason for Linalg on tensors work to begin with! (I am a bit jealous that you got to delete it before I could get to it :P ).

Just a minor comment (and making this `request changes` for that to be addressed)

**Review by hanhanW — APPROVED** [review]
Nice!

File: iree/compiler/Conversion/LinalgToSPIRV/ConvertToGPUPass.cpp (hunk 1 above)
Comment (antiagainst) [reply iree/compiler/Conversion/LinalgToSPIRV/ConvertToGPUPass.cpp]: Done.

**Comment by antiagainst** [comment]
> Awesome! I think removing `SplitDispatchFunctionPass` was the reason for Linalg on tensors work to begin with! (I am a bit jealous that you got to delete it before I could get to it :P ).



Haha. I introduced those passes to begin with so I sort of feel responsible to retire them. ;) Don't worry you'll have the fun to find and remove code; pretty sure there are more to kill!

**Review by MaheshRavishankar — APPROVED** [review]

**Pull request merged**